Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -539,74 +540,77 @@ extension SwiftLanguageServer {
     skreq[keys.sourcetext] = snapshot.text
     skreq[keys.syntactic_only] = 1
 
-    let handle = sourcekitd.send(skreq) { [weak self] result in
+    queue.async { [weak self] in
```

Comment:
I think any requests that aren't touching state protected by the `queue` can just call `self.sourcekitd.send(..., self.queue)` without doing an extra `queue.async` first.  The requests like open/edit/close, completion and cursor info currently need to touch the build settings and other state, so they need the `queue.async`.  But this request (`documentSymbol`) and I think the others below it don't, with the exception of `documentSymbolHighlight`.

Reply:
Do note that the `async`s effectively made the code run the same way that it was before (with the LSP automatically `async`ing to the `queue`), but if you think they're not needed I can remove them.